packages/palette/src/utils/useSentinelVisibility.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState, useRef } from "react"
+import { Box } from "../elements/Box"
+
+/**
+ * Used to track if the returned sentinel is in the viewport or not
+ */
+export const useSentinelVisibility = () => {
+  const [isSentinelVisible, setSentinelVisibility] = useState(false)
+  const ref = useRef<HTMLElement | null>(null)
+
+  useEffect(() => {
+    if (!("IntersectionObserver" in window)) return // Unsupported
+
+    const element = ref.current
+    if (!element) return
+
+    const observer = new IntersectionObserver(
+      ([entry]) => {
+        setSentinelVisibility(!entry.isIntersecting)
+      },
+      {
+        root: null,
```

Comment:
isn't better to extend it to make it more customizable? I would say at least the threshold and margin

Reply:
Not really, since this isn't a generic intersection observer hook. We're not using this for lazy loading or anything like that. It's a pretty specific use-case: is the sentinel in the viewport or not.